src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
     public function addPlatformToTopology(PlatformTopology $platformTopology): void
     {
         /**
-         * search for already registered central type
+         * Search for already registered central or remote top level server on this platform
+         * As only top level platform do not need parent_address and only one should be registered
          */
         if (PlatformTopology::TYPE_CENTRAL === $platformTopology->getType()) {
-            $foundCentralPlatformType = $this->platformTopologyRepository->findPlatformTopologyByType(
-                PlatformTopology::TYPE_CENTRAL
-            );
-            // search for its nagios_server ID
-            if (null !== $foundCentralPlatformType) {
+            // New unique Central top level platform case
+            $this->checkForAlreadyRegisteredPlatformType(PlatformTopology::TYPE_CENTRAL);
+            $this->checkForAlreadyRegisteredPlatformType(PlatformTopology::TYPE_REMOTE);
+            $this->setServerNagiosId($platformTopology, true);
+        } elseif (PlatformTopology::TYPE_REMOTE === $platformTopology->getType()) {
+            // Cannot add a Remote behind another Remote
+            $isLocalhost = false;
+            $this->checkForAlreadyRegisteredPlatformType(PlatformTopology::TYPE_REMOTE);
+            if (null === $platformTopology->getParentAddress()) {
+                // New unique Remote top level platform case
+                $this->checkForAlreadyRegisteredPlatformType(PlatformTopology::TYPE_CENTRAL);
+                $isLocalhost = true;
+            }
+            $this->setServerNagiosId($platformTopology, $isLocalhost);
+        }
+
+        $this->checkForAlreadyRegisteredSameNameOrAddress($platformTopology);
+        $registeredParentInTopology = $this->searchForParentPlatformAndSetId($platformTopology);
+
+        /**
+         * The top level platform is defined as a Remote
+         * Checking data consistency from 'informations' table and calling the register request on the Central
+         */
+        if ($registeredParentInTopology && true === $platformTopology->isLinkedToAnotherServer()) {
+            /**
+             * @var PlatformTopology $platformInformation
+             */
+            $platformInformation = $this->platformTopologyRepository->findPlatformInformation();
+
+            if (null === $platformInformation) {
+                throw new PlatformTopologyException(
+                    _("Platform's mandatory data are missing. Please reinstall your platform.")
```

Comment:
the exception message is a bit scary, isn't it ?

Reply:
If the `informations` table is truncated or dropped, we have no workaround and the customer needs to reinstall the centreon-web package.
This use case is the worse we can expect.
What message could I return ?